plasmapy/plasma/grids.py

```diff
@@ -1101,57 +1094,32 @@ def nearest_neighbor_interpolator(
             except AttributeError:
                 pass
 
-        # Fetch the values at those indices from each quantity
-        if self.is_uniform:
-            values = self._interp_quantities[i[:, 0], i[:, 1], i[:, 2], :]
-        else:
-            values = self._interp_quantities[i]
-
-        # Apply the NaN mask (set any values that were out of bounds
-        # to zero)
-        values *= np.outer(nan_mask, np.ones(nargs))
+        return pos, args, persistent
 
-        # Split output array into arrays with units
-        # Apply units to output arrays
-        output = []
-        for i in range(nargs):
-            output.append(values[:, i] * self._interp_units[i])
 
-        if len(output) == 1:
-            return output[0]
-        else:
-            return tuple(output)
-
-    def volume_averaged_interpolator(
-        self, pos: Union[np.ndarray, u.Quantity], *args, persistent=False
-    ):
-        r"""
-        Interpolate values on the grid using a volume-averaged scheme with
-        no higher-order weighting.
-
-        Parameters
-        ----------
-        pos : `~numpy.ndarray` or `~astropy.units.Quantity` array, shape (n,3)
-            An array of positions in space, where the second dimension
-            corresponds to the three dimensions of the grid. If a
-            `~numpy.ndarray` is provided, units will be assumed to match
-            those of the grid.
+def _fast_nearest_neighbor_interpolate(pos, ax):
+    """
+    This function finds the indices in the axis 'ax' that are closet to the
+    values in the array 'pos'
 
-        *args : `str`
-            Strings that correspond to DataArrays in the dataset
+    Assumes the axis 'ax' is sorted in accending order.
 
-        persistent : `bool`
-            If `True`, the interpolator will assume the grid and its
-            contents have not changed since the last interpolation. This
-            substantially speeds up the interpolation when many
-            interpolations are performed on the same grid in a loop.
-            ``persistent`` overrides to `False` if the arguments list
-            has changed since the last call.
-        """
-
-        raise NotImplementedError(
-            "Volume-averaged interpolator is not yet " "implemented for this grid type."
-        )
+    """
+    # Find the index where each position would be inserted into the axis.
+    # This is equivalent to a nearest neighbor interpolation but always
+    # rounding to the next highest axis point.
+    indices = np.searchsorted(ax, pos)
+
+    # searchsorted can return an out-of-bound index on the top end, so replace
+    # that point with the closest in bound index
+    indices = np.where(indices == ax.size, ax.size - 1, indices)
```

Comment:
Good call on `np.where` here, I'd have thought to use `np.clip` and that'd be 6 times slower. 👏🏻

Reply:
I rewrote this function a few times with different alternatives before arriving at this one! Might be room to speed it up even further by rewriting it with Numba...

Reply:
Probably not, because I expect numba would act on each element... Hmm. Come to think of it, my timings didn't include defining `indices` for the case of `where`, so that's cheating a little. But yeah, this looks great.